pkg/logs/input/tailer/file_provider.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package tailer
+
+import (
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+)
+
+// File represents a file we want to tail
+type File struct {
+	Path   string
+	Source *config.IntegrationConfigLogSource
+}
+
+// NewFile returns a new File
+func NewFile(path string, source *config.IntegrationConfigLogSource) *File {
+	return &File{
+		Path:   path,
+		Source: source,
+	}
+}
+
+// FileProvider implements the logic to retrieve at most filesLimit Files defined in sources
+type FileProvider struct {
+	sources    []*config.IntegrationConfigLogSource
+	filesLimit int
+}
+
+// NewFileProvider returns a new FileProvider
+func NewFileProvider(sources []*config.IntegrationConfigLogSource, filesLimit int) *FileProvider {
+	return &FileProvider{
+		sources:    sources,
+		filesLimit: filesLimit,
+	}
+}
+
+// FilesToTail returns all the Files matching paths in sources,
+// it cannot return more than filesLimit Files
+// For now, there is no way to prioritize specific Files over others
+// and the result is not meant to remain the same over serveral calls
+func (r *FileProvider) FilesToTail() []*File {
+	filesToTail := []*File{}
+	for _, source := range r.sources {
+		if len(filesToTail) == r.filesLimit {
+			log.Println("Reached the limit on the maximum number of files in use:", r.filesLimit)
+			break
+		}
+		if strings.Contains(source.Path, "*") {
```

Comment:
What is the reasoning behind restricting the supported patterns to `*` ? Since the responsibility of writing correct patterns is on the client side (and not safeguarded in our code anyway), I think we should leverage all the patterns available in `glob` so we can support more use-cases out of the box.

Reply:
Agreed, it looks like premature optimisation